R/existsBackgroundVariables.r

```diff
@@ -1,4 +1,4 @@
-existsBackgroundVariables <- function(dat, variable, warnIfMissing = FALSE )  {
+existsBackgroundVariables <- function(dat, variable, warnIfMissing = FALSE, stopIfMissingOnVars = NULL )  {
            if(!is.null(variable[1]))  {

```

Comment:
Independent of current changes: I think readability of the code would benefit substantially through using:

`if(is.null(variable[1])) return(NULL)`
`if(is.na(variable[1])) return(NULL)`

and removing `else`. This could prevent the whole function being completely indented on two levels.

Reply:
Good idea. BTW, this provokes a follow up question:  I think the function's handling of NA-arguments is inconsistent. See for example: 

library(eatTools)
data(rater)
existsBackgroundVariables(rater, c("rater", "variable"))
[1] "rater"    "variable"
existsBackgroundVariables(rater, c(NA, "rater", "variable"))
NULL
existsBackgroundVariables(rater, c("rater", NA, "variable"))
Fehler in existsBackgroundVariables(rater, c("rater", NA, "variable")) : 
  Can't find 1 variable(s) in dataset: 'NA'

Should NA values in variables argument be forbidden in general?

Reply:
I agree, I think it would be a good idea to throw an error for any `NAs` in the `variable` argument. 

Or is there any specific use case where 
`existsBackgroundVariables(rater, variable = NA) `
should return `NULL`?

Additional input validation for `dat`, `warnIfMissing` and `stopIfMissingOnVars` seems reasonable but would IMHO not be required if the function is solely used internally.

Reply:
I can't remember a specific use case where existsBackgroundVariables(rater, variable = NA) should return NULL. All right: will give error. 

dat, warnIfMissing and stopIfMissingOnVars: if a data set with UVs, AVs, weights and jackknife variables is used, missings on AV will not cause multilevel regression to crash. The user, however, might be interested if there are missings on AV, Missings ob jackknife variable, however, results in crashing BIFIEsurves::twoLevelReg with nebulous error message. (This indeed happens during the BT2022) Therefore, I would like to distinguish between crucial and noncritical missings during the eatRep check functions.